BlenderRenderController/BlenderRenderController/MainForm.cs

```diff
@@ -172,20 +245,148 @@ private void concatenatePartsButton_Click(object sender, EventArgs e)
+			p.StartInfo.UseShellExecute        = false;
+
+            p.StartInfo.Arguments = String.Format("-b \"{0}\" -P \"{1}\"",
+                                                  blendFilePathTextBox.Text,
+                                                  Path.Combine(ScriptsPath, "get_project_info.py")
+                                    );
+
+			try {
+				p.Start();
+			}
+			catch( Exception ex ) {
+				Trace.WriteLine( ex );
+			}
+
+			StringBuilder jsonInfo    = new StringBuilder();
+			bool          jsonStarted = false;
+			int           curlyStack  = 0;
+
+			while( !p.StandardOutput.EndOfStream ) {
+				string line = p.StandardOutput.ReadLine();
+
+				if( line.Contains( "{" ) ) {
+					jsonStarted = true;
+					curlyStack++;
+				}
+
+				if( jsonStarted ) {
+
+					if( !line.ToLower().Contains( "blender quit" ) && curlyStack > 0 ) {
+						jsonInfo.AppendLine( line );
+					}
+
+					if( line.Contains( "}" ) ) {
+
+						curlyStack--;
+
+						if( curlyStack == 0 ) {
+							jsonStarted = false;
+						}
+
+					}
+
+				}
+
+			}
+
+			BlenderData blendData = null;
+			if( jsonInfo.Length > 0 ) { 
+				JavaScriptSerializer serializer = new JavaScriptSerializer();
+				blendData = serializer.Deserialize<BlenderData>( jsonInfo.ToString() );
+			}
+
+			if( blendData != null ) {
+				startFrameNumericUpDown.Value      = blendData.StartFrame;
+				totalFrameCountNumericUpDown.Value = blendData.EndFrame;
+				outFolderPathTextBox.Text          = outFolderPath = blendData.OutputDirectory;
+				//blendProjectName                   = blendData.ProjectName;
+
+				if( blendData.EndFrame < endFrameNumericUpDown.Value ) {
+					endFrameNumericUpDown.Value = blendData.EndFrame;
+				}
+
+			}
+
+			Trace.WriteLine( "Json data = " + jsonInfo.ToString() );
+
+			//Trace.WriteLine( String.Format( "CEW: {0}", p.StartInfo.Arguments ) );
+
+		}
+
+		private void ReadBlenderData_Click( object sender, EventArgs e ) {
+			DoReadBlenderData();
+		}
+
+		private void MixdownAudio_Click( object sender, EventArgs e ) {
+
+			if( !File.Exists( blendFilePathTextBox.Text ) ) {
+				return;
+			}
+
+			if( !Directory.Exists( outFolderPathTextBox.Text ) ) {
+				Directory.CreateDirectory( outFolderPathTextBox.Text );
+			}
+
+            Process p = new Process();
+
+            p.StartInfo.FileName               = "blender";
+			p.StartInfo.RedirectStandardOutput = true;
+			p.StartInfo.UseShellExecute        = false;
+			//Using minimized instead so we get feedback
+			//p.StartInfo.CreateNoWindow         = true;
+			p.StartInfo.WindowStyle = ProcessWindowStyle.Minimized;
+
+            p.StartInfo.Arguments = String.Format("-b \"{0}\" -P \"{1}\"",
+                                                  blendFilePathTextBox.Text,
+                                                  Path.Combine(ScriptsPath, "mixdown_audio.py")
+                                    );
+
+			p.Start();
+
+			p.WaitForExit((int)TimeSpan.FromMinutes(5).TotalMilliseconds);
```

Comment:
This function isn't for video segments, this is for the audio.  I found that if I rendered out segments with audio it would leave silent spots at the end of the segments(AAC/AC3 audio).  Then when you concatenated them all together, you would hear clicks/pops where the segments were joined together.  The process now renders out all the segments and mixes down(user clicks a button) the audio.  Then during the concatenation of files it also muxes in the audio file.

This process (as of blender 2.76) didn't give any feedback about the audio mix down process.  So I setup this function to wait for the blender process to exit to determine if the process finished.  The timeout is better than the program hanging forever if the audio mix down never finishes.  I would like to put better error handling but with no feedback from blender, there aren't a lot of options.  Well, I mean, we should be able to check to see if we hit the timeout, but I'd like something better.  Should add it so the user is awar they need to check the output.  I've found 5 minutes to be more than enough for just the audio.

Long term, I have some plans to rework the process handling to make it more like a job queue.  Since handling the processes more generally, we can then more easily determine when rendering is done and then automatically add the concat job and add the audio mix down job.